### PR TITLE
Add conditional around the admin secrets

### PR DESCRIPTION
**Pull request by wild-endeavor**
## Tracking issue
https://github.com/flyteorg/flyte/issues/3141

## Describe your changes
https://github.com/flyteorg/flyte/pull/3002 adds the secret mount for admin to be able to other clusters in the multicluster setup but this will cause issues if there are no additional clusters.

## Check all the applicable boxes

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

### CODE REVIEW

**Comment by wild-endeavor** [comment]
@MorpheusXAUT @andrewwdye @hamersaw could you take a look at this please?

File: charts/flyte-core/templates/clusterresourcesync/deployment.yaml [hunk 1]
```diff
@@ -39,8 +39,10 @@ spec:
             name: resource-templates
           - mountPath: /etc/flyte/config
             name: config-volume
+          {{- if gt (len .Values.configmap.clusters.labelClusterMap) 0 }}
```
[the remark this answers is not in the document]
Comment (wild-endeavor) [reply charts/flyte-core/templates/clusterresourcesync/deployment.yaml:42]: i believe so yes. if you are talking to any other cluster you need this map.  The exception is if you are talking to your own cluster, in which case this is not needed.  that's my understanding at least.  @katrogan ?

[the remark this answers is not in the document]
Comment (katrogan) [reply charts/flyte-core/templates/clusterresourcesync/deployment.yaml:42]: I think you're correct @wild-endeavor. You always need credentials when the data plane is deployed on a separate cluster

[the remark this answers is not in the document]
Comment (MorpheusXAUT) [reply charts/flyte-core/templates/clusterresourcesync/deployment.yaml:42]: Correct, yes. If data and control plane are separate, you need the credentials, even if it's just one cluster.
If both are hosted together (within the "local" cluster), you can use in-cluster authentication and don't need the map/credentials.

**Review by MorpheusXAUT — APPROVED** [review]
LGTM, thanks for the fix, didn't think about the empty case.

**Review by katrogan — APPROVED** [review]

**Pull request merged**